src/Core/Form/ChoiceProvider/NonInstalledLocalizationChoiceProvider.php

```diff
@@ -26,45 +26,56 @@
 
 namespace PrestaShop\PrestaShop\Core\Form\ChoiceProvider;
 
+use PrestaShop\PrestaShop\Adapter\Language\LanguageDataProvider;
 use PrestaShop\PrestaShop\Adapter\Language\LanguageValidator;
 use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use PrestaShop\PrestaShop\Core\Language\Pack\Loader\LanguagePackLoaderInterface;
+
 
 /**
  * Class NonInstalledLocalizationChoiceProvider is responsible for getting one part of choices to use
  * in 'Improve > International > Translations' page Add / Update a language form type.
  */
 class NonInstalledLocalizationChoiceProvider implements FormChoiceProviderInterface
 {
-    /**
-     * @var array
-     */
-    private $allLocalizationChoices;
     /**
      * @var LanguageValidator
      */
     private $languageValidator;
+    /**
+     * @var LanguagePackLoaderInterface
+     */
+    private $languagePackLoader;
+    /**
+     * @var LanguageDataProvider
+     */
+    private $languageProvider;
 
-    public function __construct(array $allLocalizationChoices, LanguageValidator $languageValidator)
-    {
-
-        $this->allLocalizationChoices = $allLocalizationChoices;
+    public function __construct(
+        LanguagePackLoaderInterface $languagePackLoader,
+        LanguageValidator $languageValidator,
+        LanguageDataProvider $languageProvider
+    ) {
         $this->languageValidator = $languageValidator;
+        $this->languagePackLoader = $languagePackLoader;
+        $this->languageProvider = $languageProvider;
     }
 
     public function getChoices()
     {
-        $isEmpty = empty($this->allLocalizationChoices);
-
-        if ($isEmpty) {
-            return [];
-        }
-
+        $languages = $this->languagePackLoader->getLanguagePackList();
```

Comment:
why not make this return an `array` consistently, so you don't have to care about it being `null` later on